src/app/ReadClient.cpp

```diff
@@ -70,6 +71,26 @@ ReadClient::~ReadClient()
 
 void ReadClient::Close(CHIP_ERROR aError)
 {
+    if (aError != CHIP_NO_ERROR)
+    {
+        if (mReadPrepareParams.mShouldResubscribe)
+        {
+            CHIP_ERROR err = Resubscribe();
+            if (mReadPrepareParams.mShouldResubscribe && err == CHIP_NO_ERROR)
```

Comment:
We don't need to double-check `mShouldResubscribe` again. The err not being `CHIP_NO_ERROR` would have sufficed.

Reply:
updated, thanks